cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,59 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
-
+  beforeEach(() => {
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
+  it('should register the user', () => {
+    cy.get('.text-center')
+      .should('contain', 'Practice Form');
+
+    cy.get('#firstName')
+      .type('Ryan');
+
+    cy.get('#lastName')
+      .type('Gosling');
+
+    cy.get('#userEmail')
+      .type('fallguy24@gmail.com');
+
+    cy.get('label[for="gender-radio-1"]').click();
+
+    cy.get('#userNumber')
+      .type('0987654321');
+
+    cy.get('#dateOfBirthInput').click();
+
+    cy.get('.react-datepicker__month-select').select('November');
+
+    cy.get('.react-datepicker__year-select').select('1980');
+
+    cy.get(':nth-child(3) > .react-datepicker__day--012').click();
+
+    cy.get('.subjects-auto-complete__value-container')
+      .type('comp{enter}');
+
+    cy.get('label[for="hobbies-checkbox-1"]').click();
+
+    cy.get('label[for="hobbies-checkbox-2"]').click();
+
+    cy.get('label[for="hobbies-checkbox-3"]').click();
+
+    cy.get('#currentAddress').type('My address');
+
+    cy.get('.css-yk16xz-control > .css-1hwfws3 > .css-1wa3eu0-placeholder')
```

Comment:
Also not a very good selector. Below is also not clear. If there is no way to find anything better, then at least add a couple of comments so that it is clear what is happening everywhere.

Reply:
fixed